### ISSUE TITLE
Fedora dependencies?

### ISSUE THREAD

**Issue by otaviomad** [gonzaarcr/Fildem#36]
I'm currently running fedora 33. Are there any chances someone already found out the dependencies?

**Comment by gonzaarcr** [comment]
I couldn’t make it work on Fedora. I’ll let you know of any advances.

**Comment by otaviomad** [comment]
yep. on the official repositories I could only find the appmenu module for qt. apparently theres this: https://pkgs.org/download/appmenu-gtk3 however I dislike the idea of downloading random binaries without proper repos.

**Comment by gonzaarcr** [comment]
I did find some stuff on the repos, like `unity-gtk-module` and `appmenu-gtk-module` but couldn’t get it to work. Also, didn’t find another similar app that uses these to test.

**Comment by lfom** [comment]
Fildem probably does not work with Fedora because it default session uses Wayland. If you use XOrg instead, it should work like any other distro, provided you install the requirements, of course. It would be great to make it work in Wayland tho, at least with Qt and apps running on Xwayland, like latest KDE does.

**Comment by gonzaarcr** [comment]
This app works on Wayland. I can’t say it’s the best experience because Wayland doesn’t like floating windows and global communication, but it should at least detect some Qt menus.

**Comment by lfom** [comment]
@gonzaarcr Indeed it works! Amazing! I was searching for something like this for months! I even thought about moving to KDE, now that GTK integration is much better, but it still does not look clean enough to my taste.

On Pop!_OS 20.04 (Gnome 3.36) it didn't need `unity-gtk-module-common`, and adding `appmenu-gtk-module` to GTK_MODULES environment variable seems to be enough (I had it already set up because I did some tests with Vala panel in the past).

The HUD shortcut (Alt + space) does not work on Wayland. The only other thing that didn't work as it should was Chromium, even on XOrg: maybe it is the version I use (Ungoogled-Chromium non-portable for Debian), I will try another version later. Do you have any tips?

**Comment by lfom** [comment]
Updates:
1. HUD works on Wayland if there is no default shortcut combo pre-configured (Alt + space was set to show the window menu: maximize, minimize, etc)
2. I used Feren OS (like Ubuntu 20.04 but with latest KDE) and the same [Ungoogled Chromium version](https://ungoogled-software.github.io/ungoogled-chromium-binaries/releases/linux_portable/64bit) that I tested on Pop!_OS worked fine with the Global Menu in KDE
3. KDE in FerenOS does not use `bamfdaemon`, what surprises me. It seems that they use pure dbusmenu/appmeny libs.

**Comment by gonzaarcr** [comment]
1. The shortcut works out of the box? I created `inithud.sh` for Wayland since you cannot create shorcuts, but I forgot to document it.

**Comment by lfom** [comment]
No, my bad. Probably it worked on XOrg when I was testing. It does not work on Wayland, even with no default key combo assigned to Alt + space.
I wonder why the menus are not showing for Chromium, do you have any ideas?

**Comment by gonzaarcr** [comment]
I just tested on Zorin and it worked. Make sure you are not using the AppImage version. If you keep having troubles create an isssue.

**Comment by lestcape** [comment]
@lfom unity-gtk-module **have not support for wayland** and appmenu-gtk-module have support,  b_ut that was added by me_.  In Fedora, the package unity-gtk-module was introduced by a maintainer of the  Cinnamon desktop in Fedora, when i created a global menu for the Cinnamon desktop, but I discontinued the Cinnamon extension and then not one care about update his dependencies (i.e use appmenu-gtk-module). The version of unity-gtk-module that Fedora have is also a very old version.

https://www.reddit.com/r/Fedora/comments/d3t5gp/kde_global_menu_for_gtk_fedora_30/

**Comment by classilla** [comment]
FWIW, on Fedora 34 with Xorg and GNOME 40, https://www.linuxuprising.com/2021/02/how-to-install-fildem-global-menu-and.html was pretty much as written.

**Comment by rbreaves** [comment]
Anyone try on 36 though? It seems broken to me.
